src/orion/core/worker/trial.py

```diff
@@ -256,7 +257,9 @@ def status(self, status):
     @property
     def id(self):
         """Return hash_name which is also the database key `_id`."""
-        return self.__hash__()
+        if self.id_override is None:
+            return self.__hash__()
+        return self.id_override
```

Comment:
This hash is required to ensure uniqueness of trial parameters. If we can't use the `id` in Track for this, than we need to save the hash and use it as the unique key. Is this what happens [here]( https://github.com/Epistimio/orion/pull/289/files#diff-e326ee3c119d2fa7c784968b91906ce6R445)?

Reply:
IIRC, for legacy you just discard '_id' when you get it from the db because you can reliably recompute it using the hash function.

but for track, the _id is made unique by increasing the version tag for a given hash and the primary key is `<hash, tag>`.
so I cannot just discard the `<hash, tag>` returned by the db as I cannot guess the version tag.

Reply:
just an integer that increases if the trial has already been completed

Reply:
Is there any way this wrapper forces Orion to only use version 1 then? Because there should not be 2 trials with the same configuration in Orion.

Reply:
Track takes the latest one so Orion only sees one

Reply:
Ok so the question is, is there any mechanism to ensure that within Oríon if we try to register a trial with a configuration identical to one already in the db it will be detected and raise a DuplicateKeyError (and not update the version tag and return the latest version).

Reply:
if protocol.new_trial(*args, autoincrement=False) then no new trial is created if it already exist.
Note that the behaviour is tested in `unittests/storage/test_storage.py:test_register_duplicate_trial`